### PR TITLE
helm: Apply operator settings in configmap instead of deployment

### LINKED ISSUE #6288: Helm chart should add CSI settings to configmap instead of deployment
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The CSI settings have moved from the operator deployment to a [configmap](https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/operator.yaml#L22) for operator settings to be more easily overridden. The [helm chart](https://github.com/rook/rook/blob/master/cluster/charts/rook-ceph/templates/deployment.yaml) should have a similar change instead of continuing to use the env vars.

**Expected behavior:**
Helm chart implementation should be consistent with the non-helm install.

### LINKED ISSUE #9487: Provisioner Pods not created when using helm chart
**Deviation from expected behavior:**
None
**Expected behavior:**
The provisioner pods to be created
**How to reproduce it (minimal and precise):**
Apply the helm charts.

The rook-csi pods are not created at all, the cluster is created and the osd pods are created and complete assigning the OSDs to the physical disks but the provisioners are never created. Perhaps I am doing something wrong?

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
Some operator settings can be applied dynamically in a configmap instead of requiring the operator to restart. While these settings may be infrequently updated, applying these settings in the configmap can avoid an unnecessary operator restart. This will also make the operator deployment more consistent with the non-helm install.

**Which issue is resolved by this Pull Request:**
Resolves #9487 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: cluster/charts/rook-ceph/templates/configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,152 @@
+# Operator settings that can be updated without an operator restart
+# Operator settings that require an operator restart are found in the operator env vars
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: rook-ceph-operator-config
+data:
+  ROOK_LOG_LEVEL: {{ .Values.logLevel }}
```
Comment (BlaineEXE) [inline cluster/charts/rook-ceph/templates/configmap.yaml:8]: Should these be `{{ X | quote }}` also?

**Comment by travisn** [comment]
I've sunk long enough into fixing the formatting around this, it's just not worth continuing to put time into it. Closing as won't fix.

File: cluster/charts/rook-ceph/templates/configmap.yaml (hunk 1 above)
Comment (travisn) [reply cluster/charts/rook-ceph/templates/configmap.yaml:8]: That didn't help either

**Comment by travisn** [comment]
Taking another shot at this to make the helm settings more consistent with the non-helm operator.yaml

**Review by BlaineEXE — APPROVED** [review]

**Pull request merged**